src/test/java/uk/ac/ebi/embl/template/TemplateEntryProcessorTest.java

```diff
@@ -0,0 +1,180 @@
+        
+        TemplateVariables templateVariables = getTemplateVariables(organismName);
+        Sample sample=getSampleForTest(templateEntryProcessor,templateVariables);
+        
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(sample.getTaxId()));
+        assertEquals(sourceFeature.getScientificName(),sample.getOrganism());
+    }
+
+    public void executeEntryProcessTestWithScientificName(String taxId) throws Exception{
+        
+        TemplateVariables templateVariables = getTemplateVariables(taxId);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+        
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getScientificName(),"Homo sapiens");
+    }
+
+    public void executeEntryProcessTestWithTaxId(String taxId) throws Exception{
+
+        TemplateVariables templateVariables = getTemplateVariables(taxId);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(taxId));
+        assertEquals(sourceFeature.getScientificName(),"Homo sapiens");
+    }
+    
+    public void executeEntryProcessInvalidOrganism(String invalidOrganism) throws Exception{
+
+        TemplateVariables templateVariables = getTemplateVariables(invalidOrganism);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+
+        Collection<ValidationMessage<Origin>> messages=templateProcessorResultSet.getValidationResult().getMessages();
+        //assertTrue(messages..equals("The given Organism: \"JUNK\" is not a sample. If is added as sample please check if the sample is associated with the passed user."));
+        //Organism is not Submittable: "JUNK".
+        assertFalse(templateProcessorResultSet.getValidationResult().isValid());
+    }
+    
+    private TemplateEntryProcessor getTemplateEntryProcessor() {
+        Connection con=null;
+        return new TemplateEntryProcessor(con);
+    } 
+    
+    private TemplateVariables getTemplateVariables(String organismName){
```

Comment:
- This is template specific. Maybe we should have getTemplateVariables_ERT000002 and pass is a Function<String, TemplateVariables> to the test methods to make them generic.